Antidote/CallsManager.m

```diff
@@ -0,0 +1,331 @@
+    _allActiveCallsController = [Helper createFetchedResultsControllerForType:OCTFetchRequestTypeCall
+                                                                    predicate:predicate
+                                                                     delegate:self];
+
+    _manager = [AppContext sharedContext].profileManager.toxManager.calls;
+
+    return self;
+}
+
+#pragma mark - Public
+
+- (void)callToChat:(OCTChat *)chat
+{
+    if (self.currentCall) {
+        NSAssert(NO, @"We should not be able to make another call yet");
+        return;
+    }
+
+    OCTCall *call = [self.manager callToChat:chat enableAudio:YES enableVideo:NO error:nil];
+
+    if (! call) {
+        [[AppContext sharedContext] killCallsManager];
+    }
+
+    [self dismissAndSwitchViewControllerForCall:call];
+}
+
+- (void)handleIncomingCall:(OCTCall *)call
+{
+    if (self.currentCall && self.pendingIncomingCall) {
+        // We only support showing one incoming call at a time. Reject all others
+        [self.manager sendCallControl:OCTToxAVCallControlCancel toCall:call error:nil];
+    }
+
+    if (self.currentCall) {
+        [self notifyOfIncomingCall:call];
+        return;
+    }
+
+    [self dismissAndSwitchViewControllerForCall:call];
+}
+
+#pragma mark - RBQFetchedResultsControllerDelegate
+
+- (void) controller:(RBQFetchedResultsController *)controller
+    didChangeObject:(RBQSafeRealmObject *)anObject
+        atIndexPath:(NSIndexPath *)indexPath
+      forChangeType:(NSFetchedResultsChangeType)type
+       newIndexPath:(NSIndexPath *)newIndexPath;
+{
+
+    if (type == NSFetchedResultsChangeDelete) {
+        if ([self.allCallsController.fetchedObjects count] == 0) {
+            [[AppContext sharedContext] killCallsManager];
+        }
+    }
+
+    if ((type == NSFetchedResultsChangeInsert) && (controller == self.allActiveCallsController)) {
+        OCTCall *call = [anObject RLMObject];
+
+        // workaround for deadlock in objcTox
+        // https://github.com/Antidote-for-Tox/objcTox/issues/51
+        [self performSelector:@selector(dismissAndSwitchToActiveViewControllerForCall:) withObject:call afterDelay:0];
+    }
+}
+
+- (void)controllerDidChangeContent:(RBQFetchedResultsController *)controller
+{
+    if (controller == self.allActiveCallsController) {
+        self.currentCall = [[self.allActiveCallsController fetchedObjects] firstObject];
+        [self updateDuration:self.currentCall.callDuration];
+    }
+}
+
+#pragma mark - Updates to Current Call
+
+- (void)updateDuration:(NSTimeInterval)duration
+{
+    if ([self.currentCallViewController isKindOfClass:[ActiveCallViewController class]]) {
+        ActiveCallViewController *activeVC = (ActiveCallViewController *)self.currentCallViewController;
+        activeVC.callDuration = duration;
+    }
+}
+
+- (void)setEnableMicrophone:(BOOL)enableMicrophone
+{
+    self.manager.enableMicrophone = enableMicrophone;
```

Comment:
It's actually taken care of in 
`- (void)activeCallMicButtonPressed:(ActiveCallViewController *)controller`